### DISCUSSION TITLE
Priorities (Bigger Picture Solutions)

### DISCUSSION THREAD

**Discussion by Conza88** [bisq-network/bisq#5271] [General]
I generally love to hear about the [big picture](https://github.com/bisq-network/proposals/issues) [changes](https://github.com/bisq-network/projects/issues) that are coming, or what should at least be considered as priorities. 
The role of a CEO is in part—capital allocation. Where do they direct funds to, what projects take priority etc. Where are the biggest problems currently, and will these projects/developments fix them?

With Bisq n' the DAO, there is no CEO. So I am wondering a bit more generally, is the prioritisation set by the bounty's for these projects? With dev's taking it upon themselves to fix low hanging fruit? How about, as to whether the pieces fit together? 

For instance I've been hammering home the benefits of 5. [Increase liquidity by allowing each offer to have multiple payment methods & currencies associated with it](https://github.com/bisq-network/bisq/issues/5147) #5147 but maybe all the work would be un-done or made harder by then having to change e.g. 4 "Combining the 4 on-chain transactions into one" if that was done later on? So inefficient from an overall perspective? 

I guess I would be interested in discussions of what folks think should be prioritised? And also structure wise moving forward? Or pointing me towards something that indicates how various considerations are weighed? I've scanned the pull requests, issues, other repositories for some of (imo) top priorities. 

**1. API Release**
- Allows for Bisq integration with variety of other apps.
- Improvements to mobile app functionality? Can take offers, & make offers via mobile on the fly? Respond to traders via chat? Improves communication.
- Improved liquidity, more trades, more use.

**2. Lightning Network integration (L2)?**
- Technically hard to implement? Solutions being worked on? 
- Cheaper fee's to use Bisq, and faster transactions.

**3. Taproot & Schorr advancements?**
- Far cheaper fee's
- More privacy
- Allow's tapscript, more flexibility for smart contracts 
- Being prepared for this given its already in Bitcoin Core & activation is currently being discussed.

**4. [Reduce trade protocol to 1 single transaction](https://github.com/bisq-network/proposals/issues/279)**
- This would enable makers to place offers with no miner fees. 
- Cheaper for takers (1 trans) instead of 4.
- More liquidity as a result, given more offers & less costs, more trades
- I think a large reduction in trades on Bisq has been a result of higher fee's

**5. [Increase liquidity by allowing each offer to have multiple payment methods & currencies associated with it](https://github.com/bisq-network/bisq/issues/5147)** 
- Liquidity for sellers
- Liquidity for buyers
- True representation of smaller local markets (e.g. Transferwise/Revolut seller, and NZ offer to buy taken in NZ local market instead of... NZ local going to e.g. EUR market to buy BTC via Transferwise/Revolut). 
- More trades, local markets thrive more, more local community and word of mouth spread. 
- Traders may be willing to pay deal with higher fee's if their offers are more liquid and get taken quicker. 

**6. [Implement new-user-onboarding and new UI design](https://github.com/bisq-network/projects/issues/47)**
- [Improve "Portfolio" icon to aid communication between traders](https://github.com/bisq-network/bisq/issues/5074)
- [Add edit offer button next to the remove button in the offer book](https://github.com/bisq-network/bisq/issues/4986)
- [Fix various 'account management' usability issues in the Bisq client to optimize for traders with multiple fiat and altcoin accounts](https://github.com/bisq-network/bisq/issues/4837)
- [Improve 'Receive payment' screen](https://github.com/bisq-network/bisq/issues/4721)
- [On "Send Funds" screen add ability to change withdrawal trans fee sats rate](https://github.com/bisq-network/bisq/issues/5199)
- [Show TOTAL fees in the "Fund your offer" message box, also in local currency](https://github.com/bisq-network/bisq/issues/4938)
- Easier more welcoming walk-through UX, more likely to keep new users engaged, more likely to trust what they are doing & others, thus more trades and users.  

**[7. Improving Support Mediation](https://github.com/bisq-network/projects/issues/48)**
- [Add ability for mediator's & arbitrator's to see "Trader Chat" (upon escalation)](https://github.com/bisq-network/bisq/issues/5044)
- [A new mediation resolution doesn't reset the trader's choices](https://github.com/bisq-network/bisq/issues/5221)
- [Trade closed by arbitrator won't disappear from "Open Trades" tab](https://github.com/bisq-network/bisq/issues/5150)
- [Add a placeholder on the "Search Disputes" field](https://github.com/bisq-network/bisq/issues/5105)
- [Add Notification icon for Mediation and Arbitration chat](https://github.com/bisq-network/bisq/issues/5097)
- [Promote Keybase as a support channel within Bisq GUI](https://github.com/bisq-network/bisq/issues/4904)
- [Show mediation summary to the trader who did not open arbitration](https://github.com/bisq-network/bisq/issues/5062)

**8. Improve BSQ overall**
- [Implement Segwit for BSQ](https://github.com/bisq-network/bisq/pull/5000)
- [Improve BSQ account creation & purchase process](https://github.com/bisq-network/bisq/issues/5046)
- [Improve BSQ accessibility through OTC market and bonded reputation](https://github.com/bisq-network/proposals/issues/308)
- [Add Low priority payment type to BSQ](https://github.com/bisq-network/bisq/issues/5153)
- [BSQ balance on wallet info screen includes merit BSQ](https://github.com/bisq-network/bisq/issues/5013)
- [BSQ Fees equivalent not correctly calulated into BTC in info tooltip](https://github.com/bisq-network/bisq/issues/4959)
- Very poor UX with buying BSQ. BSQ should not be in the 'alt-coin' camp either imo. Coloured BTC is better. Make more present/clear in app (without implying its better than BTC). 

[**9. Improve Reporting**](https://github.com/bisq-network/bisq/discussions/5171)
- [Add a further column/s to make "Portfolio Performance" easier to determine](https://github.com/bisq-network/bisq/issues/5069)
- [Running the numbers on Bisq DAO revenues](https://github.com/bisq-network/bisq/discussions/5171)
- [Fix various 'account management' usability issues in the Bisq client to optimize for traders with multiple fiat and altcoin accounts](https://github.com/bisq-network/bisq/issues/4837)
- Help traders determine how their overall history and trades are going. Helps those assess BSQ performance and health of Bisq.

**10. [Additional universal fiat payment methods](https://github.com/bisq-network/growth/issues)**
- Priority on Fintech's similar to (Transferwise/Revolut) [universal fiat to many countries]
- SWIFT etc. 
- Then perhaps catering to most used markets e.g. Nigeria, India etc. those being banned and need Bisq.

**10. [Using Bisq for Peer to Peer BTC loans and interest accounts](https://github.com/bisq-network/bisq/discussions/5213)**
- Likely some opportunity to do so, but I think after all the other functions are improved. Also more likely next cycle imo demand.

Bug squashing and reliability is obviously a given but a department I lack knowledge about. In any case, please jump in with priority preferences, or if there's a road map I've missed etc.

**Comment by wallclockbuilder** [comment]
In a day where bisq is doing revenues of $500k per day($20m per month) whereas binance is doing $30Billion per day, its clear bisq is overwhelmingly under-performing. The upside is huge if bisq can manage to be customer-centric and user friendly. Right now its just a tight community of anonymous crypto hobbyists.
Binance Nigeria alone was doing about $5Billion volume daily before the ban.

To enjoy widespread success, Bisq needs to measure the project's access-ability from the newb user's perspective. How many minutes can they go from I want bitcoin to I got bitcoin? Benchmark against the centralized options: binance etc. Refine and improve until bisque is objectively faster.

It will be interesting to see what percentage of users download the application, run it and bounce because its too many steps vs the percentage who stick to their guns until they do their first trade.
How long does it take most people to go from download to first successful trade? How long does it take them to quit? These stats can give some insight into end-user pain points which the growth team can focus on.

**Reply by RoninShinobi** [reply]
''Binance Nigeria alone was doing about $5Billion volume daily'' WOW

**Comment by flix1** [comment]
Low-hanging fruit potential growth drivers:

**1. Stablecoins**
Stablecoins like USDT drive huge volumes on centralized exchanges. Bisq should make an effort to start up these markets. The effort required is low and potential reward is huge. See https://github.com/bisq-network/proposals/issues/170

**2. Privacycoins**
Many are banned from centralized exchanges, others lose their privacy when trading on cex. Bisq has a natural advantage in this market. You only need to have sufficient liquidity to compete. We see that it is possible with XMR.

**3. Mobile app**
The API should allow many types of mobile apps to connect to Bisq. Feedback from many countries (Nigeria, China, India, Argentina) tells us that being desktop only is a huge UX hurdle. Making a mobile app is easy and cheap. We can build on top of this: https://docs.bisq.network/bisq-mobile.html

**Reply by Conza88** [reply]
I agree good avenues to pursue. I'd personally be low prioritising them, over everything else focused on getting the basics right.
Can we confidently recommend Bisq to a new-coiner? Unfortunately not _yet_.

**Comment by flix1** [comment]
The above are immediate (ie: 2021) potential growth drivers that would require little effort and should IMHO be short-term priorities.

For longer term growth, if we really want to compete on an equal footing with centralized exchanges we have to re-think who Bisq's target market is and TALK TO THEM, engage with them, gather feedback on why they are not using Bisq. I can imagine many of the reasons (liquidity, UX, trade time, etc) but to do this properly you need to hear it from them!

That being said, here are some of my ideas for long-term priorities:

**1. ATOMIC SWAPS**
I think the future for altcoin-BTC trading will be **atomic swaps**. It's already in the works for XMR (an important market for Bisq). It will happen for most altcoins. But even if trade execution is an atomic swap, you still need a marketplace to set price, make offers and find sellers. Bisq could be that marketplace. 

**2. BSQ and other coloured coins**
BSQ can do even better than atomic swaps. **ONE TX exchange** between BTC-BSQ is possible and has been discussed before. We should absolutely do it not just for the direct advantage, but because it can also be used as a standard for many, many possible **bitcoin coloured coin** templates. The Bisq coloured coin protocol could become a popular way to issue tokens and compete with ERC20, omni, etc...  (see https://github.com/bisq-network/proposals/issues/50)

**3. CENSORSHIP RESISTANCE**
Bisq's biggest strength is censorship resistance. We should leverage it. We should make a real effort to reach out to markets where other players have been banned. China, India, Nigeria, Argentina, Venezuela... there is a long list of countries where there are some kind of restrictions or bans on centralized exchanges, or F2F cash exchanges, or any bitcoin trading at all... If Bisq is unable to compete in these markets where most of the competition has been pushed out... then what the hell are we doing? What is the point of Bisq at all? We should make a big effort to find out how to break into these markets. 

The Chinese experience is interesting (see https://github.com/bisq-network/growth/issues/135 )

**4. MARKET MAKERS**
You cannot have constant liquidity in a P2P exchange without professionals. They keep the market alive 90% of the time, while basic users come and go. Bisq has failed to attract market makers, with a few exceptions. Look to Localbitcoins and Paxful for how to successfully attract them. They are key to kick-starting a market because new users who see an empty offer book will just leave. We need to actively engage with existing market makers on other platforms and get their feedback on why they are not using Bisq. There have been several experimental bounties on this 
(see https://github.com/bisq-network/growth/issues/135  https://github.com/bisq-network/growth/issues/184   https://github.com/bisq-network/growth/issues/138 ) with varying degrees of success, but we have not cracked how to really do this yet...

I think the API will help a lot with this. But until we solve it, the market maker problem it should be a priority.

**5. ONBOARDING and UX**
New user experience has improved significantly, but it is still horrible. Bisq is only for advanced users. We need a **UX teardown** from someone OUTSIDE Bisq (see https://www.useronboard.com/user-onboarding-teardowns/) who can objectively help us improve. It CANNOT come from devs or existing project leaders. It has to be someone with UX experience coming to it with fresh eyes. I could not do it, I already know how to use Bisq and now find it easy... but newbys are overwhelmed.

If Bisq is unable to attract pros (market makers) and also too difficult for noobs... what market does that leave us?

**Reply by Conza88** [reply]
Support 1, 2, 3... above but 4 has most importance imo — and how to attract/keep them is their offers being taken quickly & low fee's.
This translates to having more users available to them to take their offers i.e. from OP:

5. Increase liquidity by allowing each offer to have multiple payment methods & currencies associated with it
6. Implement new-user-onboarding and new UI design (which you raise above). Stuff in the works already, work just needs to be done/started. 

_But until we solve it, the market maker problem it should be a priority._

I agree.

**Comment by m52go** [comment]
There's no doubt onboarding and UX can be improved a lot, but much of the work to do this has already been done. [Pedro's prototypes](https://xd.adobe.com/view/b01dfd7a-3f79-4744-8df9-08394d2ea1ea-e54e/) would go pretty far to achieve this. Problem is getting a skilled JavaFX developer to actually implement them...funding and talent need to be procured.

Another more immediate issue is [reducing minimum required deposits](https://github.com/bisq-network/proposals/issues/295).

> If Bisq is unable to attract pros (market makers) and also too difficult for noobs... what market does that leave us?

I had a passing correspondence with a vendor who was earning BTC revenue and was having issues with one of the big exchanges in selling the BTC for fiat. I'm not sure Bisq would be a viable option for this use case without the API, but will follow up anyway...there must be more like them.

> In a day where bisq is doing revenues of $500k per day($20m per month) whereas binance is doing $30Billion per day, its clear bisq is overwhelmingly under-performing.

Binance is engineered for convenience at the expense of everything else, and Bisq is engineered for freedom at the expense of everything else. Of course Bisq can do better than it's currently doing but it's unreasonable to compare its volumes to operations with vastly different goals. Apples and oranges.

For growth I can say there _was_ a project board established last year for prioritization but there really wasn't anyone around to collaborate with so I stopped maintaining it.

China needs a mobile app. India seems workable without one, but [they need UPI](https://github.com/bisq-network/growth/issues/225) as a payment method.

**Reply by Conza88** [reply]
> Problem is getting a skilled JavaFX developer to actually implement them...funding and talent need to be procured.

How is BSQ going in terms of usage? Trades for it are real low yeah? Focusing improving that process ability to buy BSQ (and thus sell for BSQ holders) should be a priority? ATM massive barriers in the way of people being able to buy simply. 

This would help funding overall? Lifeblood of project?

**Reply by philippnagel** [reply]
@m52go why does China need a mobile app? Because the users there are mobile or to crack the Great Firewall as opposed to a web app?

**Reply by m52go** [reply]
Users greatly prefer mobile and find the desktop program clunky -- that's the feedback we got from folks on the ground who tried to bootstrap the market in the past. If I remember correctly, I think they also mentioned that circumventing GFW is easier on mobile than on desktop.

**Reply by ripcurlx** [reply]
I guess it is easier to use a luxury hotel wifi with a smartphone than by sitting in the lobby with your laptop 😜

**Comment by sqrrm** [comment]
The lack of a CEO type of entity within Bisq is something that has been discussed before and no good solutions have been found. I think the closest model we work after is Bitcoin where everyone works on what they see fit and only very small changes can be done, and when done, very slowly since a rough consensus has to be found before moving ahead.

In Bisq we have the benefit of the voting to move decisions forward much faster, but we lack the resources and direction. Many of the point suggested here are not even contentious, just a lot of work. I think it's fine that everyone works on what they want, but it is a problem that progress is slow, many times due to the bisq client being monolithic. It makes it very hard to change anything without causing trouble for other developers in other domains and it makes it slow to move forward with dev work. Now that the API is ready I think some points could be worked on much more independently, moving certain projects forward without depending on some central reviewer or other issues stemming from the complex nature of the bisq client.

@flix1 As an aside I can mention that I'm trying to get the BSQ atomic swaps working. It touches almost all modules, p2p, dao, trade protocol, offer protocol, desktop. It would also not be easy to extend the atomic swap functionality without dependence on a wallet for the asset that needs to be swapped, how else do you verify your tx?

To get a bit more direction perhaps it would be good for a non dev to partner with a dev for work that needs to be done. Especially good if the non dev contributors would act a bit as a patron to new devs coming in.

**Reply by flix1** [reply]
@sqrrm I am very interested in this

>  I'm trying to get the **BSQ atomic swaps** working. It touches almost all modules, p2p, dao, trade protocol, offer protocol, desktop.

I'll get in touch with you on keybase to see how I can help.

> It would also not be easy to extend the atomic swap functionality without dependence on a wallet for the asset that needs to be swapped, how else do you verify your tx?

I would not worry (too much) about other assets right now. The priority should be to prove that BSQ-BTC atomic swaps (a.k.a one-tx exchange) are possible and get them working.  If it can be done for one coloured coin, it can be done for others built on the same template.

**Comment by ripcurlx** [comment]
@Conza88 Thanks for your compilation of priorities! I think it contains most of the challenge we need to tackle. Before getting into the resources allocation issue here is my personal priorities list for now.

## Operational priorities (need to be tackled)
- Bug fixes
- Upgrade Java/JavaFX (to get required security patches) #4242 
- Move nodes to Tor v3 until 3rd October 15th, 2021 (https://blog.torproject.org/v2-deprecation-timeline)

We do have a [priorities board](https://github.com/orgs/bisq-network/projects/18?fullscreen=true) which covers the current challenges, but no further roadmap. This is mainly because of the lack of resources. It is great to have a long-term roadmap, but we just need devs to be able to tackle them.

## Priorities
- API
- Improve reporting
- Reduce or eliminate mining fees wherever possible (BSQ, trade protocol,...)
- Improve onboarding

I think what Bisq needs is a small team (4-6) of full-time contributors, who steer the ship, so other contributors can come and go doing stuff for Bisq every now and then. I wasn't able to contribute more than to review/merge and ship new versions in the last couple of months because of my paternity leave which will continue until summer. I should be able to contribute more time to Bisq again afterwards. 

I personally think what we need to do is first to enable contributors to make changes to Bisq and to be able to see if it has a positive effect or not. To enable that the whole Bisq monitor project was started that never reached a stable state and was recently taken over by @Emzy (https://monitor.bisq.network/d/dEVJDduWz/key-performance-indicators?orgId=1). It has with `offers per node` and `trades per offer` two metrics that could be used to make comparisons between app versions if we improved the status quo or not. It is not perfect and has no control groups, but as we need to protect everyone's privacy it is a good first start. Unfortunately it is not working right now and I need to find the time to look into the implementation/setup to get it up and running again. 

A good first step could be as @sqrrm mentioned to establish small project teams of new devs and long-time non-dev Bisq contributors. Maybe we could come up with a list of non-dev contributors who are willing to join such project teams so they can be matched with new devs coming in.

**Reply by Conza88** [reply]
I'm definitely in the dark on operational priorities, but certainly see them as important. Reducing hassles for support I know. Naturally they should be a guidepost for what bug fixes are causing the most hassles/issues. 

Definitely think a core bunch of dedicated dev's to steer the ship is required. Fund wise - is this the real hold up? Surely some whales out there, or Jack & JayZ (lol) could support Bisq (i.e. the GLOBE) and not just Africa with a dev fund (not denigrating that at all btw). 

Are BSQ bought for trades not enough funding? How long till the payouts to those who got the bad end of the stick on the 'security breach' way back? That's slowing things down? 

API - allowing people to help integrate, freedom is sweet. 
Reporting - been seeing steps in that direction which has been great. 
And yeah minimising fees (increasing volume then), as well as on-boarding. Agree with all that.

**Comment by flix1** [comment]
The whole CEO concern I think is not such a big deal as we make it out to be. We are perhaps a bit too influenced by what we have seen in other startups. If Bisq has a clear **mission**, priorities can just as easily be set by a small group as by a CEO. You can have a structure like this:

------------
0. **DAO**                                       --> votes on compensation and important decisions
1. **Team Leads** (5-10)                 --> collegiate decisions on priorities, individual direction of regular work
2. **Regular contributors** (10-50) --> work under team leads
3. **Occasional contributors** (100) --> join projects under team leads or work for bounties
4. **Market makers**, power users (1000) --> have their own business models within Bisq but are fully independent
5. **Users** (10.000-100.000) --> Use Bisq. Can participate in DAO.
------------

A council of 5-10  **Team Leads** should be able to agree on priorities most of the time, or break ties through majority vote or in extreme cases put it to the DAO for a vote. 

Of course right now Bisq is even smaller than the minimum numbers described above. Most of the discussion about structure is moot. The problem is not how to organize the people we don't have! The problem is how to attract more contributors and how to pay them. GROWTH should be the priority.

![image](https://user-images.githubusercontent.com/4233784/110772824-546d8d80-825c-11eb-8b95-8aea41f31579.png)

With the current economics I don't see how Bisq can ever afford more than 2-3 full time devs. You don't need a CEO or any org structure with 3 devs. What you do need is to be asking yourself every single day: 

**"How do we double trading volumes in the next 6 months?"**

**Reply by Conza88** [reply]
I agree with that, and wasn't suggesting need for CEO or anything. Just the heightened understanding that in organisations, that position / person has a very important role... and if those actions don't get performed or thought about at all... that's not necessarily a good thing. 

Which is where you get lots of little things being snapped up, but major projects that are far more important go waiting/stagnant etc. Such is my concern anyway. 

**"How do we double trading volumes in the next 6 months?"**

That's I think what needs to be addressed yes, and what I was driving at with summarising issues. Top things that shouldn't be as 'next level as Taproot & Schorr etc.'

**1. API Release
4. Reduce trade protocol to 1 single transaction
5. Increase liquidity by allowing each offer to have multiple payment methods & currencies associated with it
6. Implement new-user-onboarding and new UI design
8. Improve BSQ buying/selling overall process**

**Reply by pazza83** [reply]
> "How do we double trading volumes in the next 6 months?"

Hi @Conza88 thanks for creating this thread. I would like the question to be specific to BTC. 

"How do we double trading volumes in BTC the next 6 months?"

BTC, over time, is going to do it's thing and keep increasing in price. It would be easy to look at trade volumes in USD and say  well done, when in reality if the volume of BTC traded is falling I wouldn't see that as a success.

In answer to how do we double trading volumes in BTC the next 6 months? I would suggest 6 months is a short period therefore a quick win is needed.  If Bisq doubles its XMR/BTC volume in 6 months it will achieve this objective and can be done without any need for valuable dev resources that can go towards the points you have discussed. 

Therefore, I would pose the question how can we double XMR/BTC volume in 6 months?

**Comment by Conza88** [comment]
So:

- [x] 1. API Release
- [ ] 4. Reduce trade protocol to 1 single transaction
- [ ] 5. Increase liquidity by allowing each offer to have multiple payment methods & currencies associated with it
- [ ] 6. Implement new-user-onboarding and new UI design
- [x] 8. Improve BSQ buying/selling overall process

1. Done (to an extent?)
4. Stalled @sqrrm , waiting for someone to check work: https://twitter.com/Conza/status/1418724944234442753
https://github.com/bisq-network/bitcoinj/pull/37
5. No
6. No
8. No

----------------
7. Improve support mediation
- Recently had my first experience of this in a long time.
- Title of mediation chat not clear (am I speaking to mediator & trader in same chat?)
- No comment from mediator at all before closing, no explanation or engagement. 
- The pop-up to accept or not was on the spot, did not appear I could close and come back or find more info out. I ended up accepting. 
- The message after that appeared to say the other had DECLINED and was seeking arbitration. Would have to wait for X days etc. 
- Turns out he had not yet responded. I was irked in the middle. There was no pop up about he had accepted or anything, just a 'closed' message pop-up. 
- I don't know what that then meant, until I had to seek others out and confirm. 
- The original message of payout was not clear either, was I being penalised at all? Where in the world is the communication - clear - that the trade is going through as it should, and no penalty to you? Not everyone spends their days interpreting trade info etc. 
- No customer satisfaction captured (CSAT). Agents incentivised to close tickets, not be clear and communicate, and turn around issues for traders.

**Comment by pazza83** [comment]
Sharing the[ Bisq DAO summary report](https://docs.google.com/document/d/1MxajknrXxZvJ4MKMhtaVhXfkb6TlaINTbqmryCbCUbw/edit) commissioned by @chimp1984 as it feels a relevant place to share it.

**Comment by Conza88** [comment]
- [x]  1. API Release
- [ ]  4. Reduce trade protocol to 1 single transaction
- [x]  5. Increase liquidity by allowing each offer to have multiple payment methods & currencies associated with it
- [ ]  6. Implement new-user-onboarding and new UI design
- [x]  8. Improve BSQ buying/selling overall process

Would we say 8. has been ticked off? Atomic swaps for BSQ - very noice. 

Still incredibly adamant that if Bisq dev's are able to get at least 4. and especially 5 - ticked off, before big moves/time spent on Misq (Bisq 2), then a shite load of bootstrapping can happen organically, when others realise their country DOES have folks selling/buying... and its not all just in one market mostly EUR. 

That and single transaction (lower fees), oooof, come at me.

**Reply by MwithM** [reply]
I still miss the "buy BSQ" clearly visible button, or at least display the BSQ available balance on the top.

**Reply by Conza88** [reply]
> I still miss the "buy BSQ" clearly visible button, or at least display the BSQ available balance on the top.

Yes, absolutely. There were several pr's that basically did ask that from memory? UX that got sidelined I guess because of the swap or coming overhaul?

Should definitely want to highlight in a way that doesn't come off as it being a shitcoin, or in your face that is off putting but it definitely needs to be present somewhere more prominent imo.

**Reply by ripcurlx** [reply]
`Buy BSQ` is only displayed if you don't have enough BSQ at hand to pay for the fee

**Reply by MwithM** [reply]
Never happened to me 😎

**Reply by Conza88** [reply]
Haha ok. That makes sense.

**Comment by Conza88** [comment]
Remaining to date:

 4. Reduce trade protocol to 1 single transaction
 6. Implement new-user-onboarding and new UI design

4 being easily the most important by far.
